src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +46,54 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
-    return false;
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
+  }
+
+  async function handleProfileUpdate() {
+    await updateUser({ ...getPayload() }, () => {
+      setActiveTab(Tabs.SUCCESS);
+      snackbarContext.show({
+        message: "Congratulations, your volunteer account has been created!",
+        type: "success",
+      });
     });
-    updateUser();
-    setActiveTab(Tabs.PROFILE);
-    return false;
   }
-  async function updateUser() {
-    const payload = getPayload();
 
-    let location = payload?.location;
+  async function updateUser(payload, callback) {
     try {
-      if (location.address) {
-        const data = await addressLookup(location.address);
-        location = { ...location, lat: data.lat, lng: data.long };
-      }
```

Comment:
Just so it's clear, this could be removed because the component used in the profile section fills this data in (and `addressLookup` wasn't working anyway)